__init__.py

```diff
@@ -4,50 +4,133 @@
 import os
 import shutil
 import sublime
+import importlib
 
-lib_path = os.path.join(os.path.dirname(sublime.packages_path()),"Lib","python38")
-
-dependencies = ["aenum","dateutil","pieces_os_client","pydantic","urllib3","websockets","six.py","typing_extensions.py"]
+__version__ = "0.1"
 
-def download_github_repo(url, path):
-    if not url.endswith('/'):
-        url += '/'
+repo_dependencies = [
+	{
+		"dependency":"aenum",
+		"version":(3,1,15)
+	},
+	{
+		"dependency":"dateutil",
+		"version":"2.9.0.post0"
+	},
+	{
+		"dependency":"pieces_os_client",
+		"version":"1.2.7"
+	},
+	{
+		"dependency":"pydantic",
+		"version":"1.10.13"
+	},
+	{
+		"dependency":"urllib3",
+		"version":"2.2.1"
+	},
+	{
+		"dependency":"websockets",
+		"version":"12.0"
+	},
+	{
+		"dependency":"six.py",
+		"version":"1.16.0"
+	},
+	{
+		"dependency":"typing_extensions.py",
+		"version":"unknown"
+	}] # Need to be changed each release
 
-    url += 'archive/master.zip'
 
-    response = urllib.request.urlopen(url)
-    z = zipfile.ZipFile(io.BytesIO(response.read()))
+lib_path = os.path.join(os.path.dirname(sublime.packages_path()),"Lib","python38")
 
-    # Create a temporary directory to extract files to
-    temp_dir = os.path.join(path,"TEMP_PIECES")
-    z.extractall(temp_dir)
 
-    # Get the name of the directory that was created in the zip
-    project_dir = os.path.join(temp_dir, os.listdir(temp_dir)[0])
 
-    # Move the files from the project directory to the desired path
-    for filename in os.listdir(project_dir):
-        try:
-            shutil.move(os.path.join(project_dir, filename), path)
-        except:
-            pass
+def download_github_repo(url, path):
+	if not url.endswith('/'):
+		url += '/'
+
+	url += 'archive/master.zip'
+
+	response = urllib.request.urlopen(url)
+	z = zipfile.ZipFile(io.BytesIO(response.read()))
+
+	# Create a temporary directory to extract files to
+	temp_dir = os.path.join(path,"TEMP_PIECES")
+	z.extractall(temp_dir)
+
+	# Get the name of the directory that was created in the zip
+	project_dir = os.path.join(temp_dir, os.listdir(temp_dir)[0])
+
+	# Move the files from the project directory to the desired path
+	for filename in os.listdir(project_dir):
+		file_path = os.path.join(project_dir, filename)
+		try:
+			shutil.move(file_path, path)
+		except:
+			file_to_remove = os.path.join(path,filename)
+			if os.path.isfile(file_to_remove): # Remove a file
+				os.remove(file_to_remove)
+			else: # Remove the folder
+				shutil.rmtree(file_to_remove)
+			shutil.move(file_path, path)
+
+	# Remove the temporary directory
+	shutil.rmtree(temp_dir)
```

Comment:
this is okay just I think it would have been a lot easier to do this via shell commands